### PR TITLE
Vyos snmp_server Resource Module

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY

Added vyos_snmp_server resource module

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ Name | Description
 [vyos.vyos.vyos_ping](https://github.com/ansible-collections/vyos.vyos/blob/main/docs/vyos.vyos.vyos_ping_module.rst)|Tests reachability using ping from VyOS network devices
 [vyos.vyos.vyos_prefix_lists](https://github.com/ansible-collections/vyos.vyos/blob/main/docs/vyos.vyos.vyos_prefix_lists_module.rst)|Prefix-Lists resource module for VyOS
 [vyos.vyos.vyos_route_maps](https://github.com/ansible-collections/vyos.vyos/blob/main/docs/vyos.vyos.vyos_route_maps_module.rst)|Route Map Resource Module.
+[vyos.vyos.vyos_snmp_serve](https://github.com/ansible-collections/vyos.vyos/blob/main/docs/vyos.vyos.vyos_snmp_serve_module.rst)|Manages snmp_server resource module
```
Comment (NilashishC) [inline README.md:63]:
```suggestion
[vyos.vyos.vyos_snmp_server](https://github.com/ansible-collections/vyos.vyos/blob/main/docs/vyos.vyos.vyos_snmp_serve_module.rst)|Manages snmp_server resource module
```

File: docs/vyos.vyos.vyos_snmp_serve_module.rst [hunk 2]
```diff
@@ -0,0 +1,1313 @@
+.. _vyos.vyos.vyos_snmp_serve_module:
```
Comment (NilashishC) [inline docs/vyos.vyos.vyos_snmp_serve_module.rst:1]: Module name needs to be fixed.

File: plugins/modules/vyos_snmp_server.py [hunk 3]
```diff
@@ -0,0 +1,1126 @@
...
+  type: dict
+  sample: >
+    This output will always be in the same format as the
+    module argspec.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: when I(state) is C(merged), C(replaced), C(overridden), C(deleted) or C(purged)
+  type: list
+  sample:
+    - sample command 1
```
Comment (ashwini-mhatre) [inline plugins/modules/vyos_snmp_server.py:1066]: Can you add valid command?

File: plugins/modules/vyos_snmp_server.py [hunk 4]
```diff
@@ -0,0 +1,1126 @@
...
+    - sample command 2
+    - sample command 3
+rendered:
+  description: The provided configuration in the task rendered in device-native format (offline).
+  returned: when I(state) is C(rendered)
+  type: list
+  sample:
+    - sample command 1
```
Comment (ashwini-mhatre) [inline plugins/modules/vyos_snmp_server.py:1074]: Same as above

**Review by ashwini-mhatre — APPROVED** [review]

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Pull request merged**